File: frontend/src/shared/ids.ts
import { v7 as uuidv7 } from "uuid"

/**
 * 主键由前端生成，不向服务器申请。
 * 用 v7 而不是 v4：v7 前 48 位是毫秒时间戳且大端序，
 * 存进 MySQL 的 BINARY(16) 后天然按时间递增，插入不会造成 B+ 树页分裂。
 */
export function newId(): string {
  return uuidv7()
}

File: backend/app/core/ids.py
import uuid

import uuid_utils

def new_id() -> uuid.UUID:
    """生成 UUID v7。注意：业务主键正常由客户端生成，这里主要给测试和服务端建的记录用。"""
    return uuid.UUID(str(uuid_utils.uuid7()))
